sources/presentation/Xenko.Core.Presentation/ValueConverters/ObjectToTypeName.cs

```diff
@@ -19,11 +22,22 @@ public class ObjectToTypeName : OneWayValueConverter<ObjectToTypeName>
         /// </summary>
         public const string NullObjectType = "(None)";
 
+        private readonly CSharpCodeProvider codeProvider = new CSharpCodeProvider();
+
         /// <inheritdoc/>
         [NotNull]
         public override object Convert(object value, Type targetType, object parameter, CultureInfo culture)
         {
-            return value?.GetType().Name ?? NullObjectType;
+            if (value == null) return NullObjectType;
+
+            Type valueType = value.GetType();
+            var fullTypeName = codeProvider.GetTypeOutput(new CodeTypeReference(valueType));
+
+            //Change System.Nullable to "Type?" format
+            var typeName = Regex.Replace(fullTypeName, @"System\.Nullable<([^>]*)>", "$1?");
```

Comment:
Will this work with `Nullable<Dictionary<int, int>>` ?

Reply:
Not sure, that's why we need unit tests. ;)

I will add tests for nested Generic types.

Reply:
I might have to get crazy with regex. https://docs.microsoft.com/en-us/dotnet/standard/base-types/grouping-constructs-in-regular-expressions#balancing-group-definitions

I might do similar for `System.ValueTuple`. Not sure how well such types work in the Property grid though.

Reply:
I wouldn't recommend playing with string regex for that.
We have a proper source `Type` so use this as source rather than reprocessing a string and doing it in two steps (esp. if the second step could be quite error-prone, we almost need a parser to properly understand generic depth, etc.).

Did you check if there wasn't a library that could do that automatically for us?
FYI we already link Roslyn (Microsoft.CodeAnalysis).

Reply:
My initial searching got me to `CSharpCodeProvider.GetTypeOutput` method. I'll do a bit more research.

Reply:
Re-wrote ObjectToTypeConverter to use a custom method that handles nullables and tuples better